packages/webdriverio/src/commands/browser/switchWindow.ts

```diff
@@ -64,6 +64,14 @@ export default async function switchWindow (
         if (matchesTarget(title)) {
             return tab
         }
+
+        /**
+         * check window name
+         */
+        const windowName:string = await this.execute('return typeof(window) !== "undefined" && window.name')
```

Comment:
Move to private method `getWindowName`

Reply:
It was implemented like that first with a getwindowname method but was suggested to be avoided and made as a single liner

Reply:
Oh that's strange because the other implementations do use that so it would be good to have it aligned imo. If however there's a good reason not to do it like that, I would love to hear why :)

Reply:
@erwinheitzman https://github.com/webdriverio/webdriverio/pull/7197 could you have a look at this discussions , i have made bit changes now . Could you have a look at why the unit test failing . I am not able to understand how the mock is implemented

Reply:
I think it goes over the list of values until it matches something, at which point it will probably fetch the correct window (probably also mocked).

I'm making quite some assumptions here but since this is a branch on your personal account I can't access it and run it locally so hopefully this helps

Reply:
It's a open for feel free to clone and have a look , I am really stuck there not able to understand . 

In real situation the code works fine , for mock it's failing